monitor/plugins/spark_progress.py

```diff
@@ -88,37 +89,32 @@ def _publish_measurement(self, job_request):
                 ref_value = (elapsed_time / self.job_expected_time)
 
                 # Error
-                error = job_progress - ref_value
-                
+                if self.job_expected_time < 0.0:
+                    error = 1.0
+                else:
+                    error = job_progress - ref_value
+
                 application_progress_error['name'] = ('application-progress'
                                                       '.error')
 
                 application_progress_error['value'] = error
                 application_progress_error['timestamp'] = time.time() * 1000
                 application_progress_error['dimensions'] = self.dimensions
 
-                print application_progress_error['value']
-
                 self.monasca.send_metrics([application_progress_error])
 
             time.sleep(MONITORING_INTERVAL)
 
 
     def _get_elapsed_time(self, gmt_timestamp):
-        try:
-            local_tz = tzlocal.get_localzone()
-
-        except Exception as e:
-            local_tz = "America/Recife"
-            local_tz = pytz.timezone(local_tz)
+        submission_date = datetime.strptime(gmt_timestamp,
```

Comment:
Does this change resolve the problem with different timezones?

Reply:
The problem with different timezones is solved when the two hosts (WP3 infrastructure and cluster) is synchronized. These sync must be done in deploy stage.